src/axolotl/utils/chat_templates.py

```diff
@@ -17,20 +44,67 @@ def chat_templates(user_choice: str):
-    }
-
-    if user_choice in templates:
-        return templates[user_choice]
+        user_choice = user_choice[
+            len(_DEFAULT_FALLBACK_CHATML_TEMPLATE_CHOICE_PREFIX) :
+        ]
+        LOG.warning(
+            f"No chat template found on tokenizer, falling back to {user_choice}. It is recommended to set --train_on_inputs to True for the model to learn this chat template."
+        )
+
+    if user_choice in _TEMPLATES:
+        return _TEMPLATES[user_choice]
 
     raise ValueError(f"Template '{user_choice}' not found.")
+
+
+def extract_chat_template_args(cfg, ds_cfg: Optional[Dict[str, Any]] = None):
+    if ds_cfg and ds_cfg.get("chat_template"):
+        chat_template_choice = ds_cfg.get("chat_template") or "chatml"
+        chat_template_jinja = ds_cfg.get("chat_template_jinja")
+    else:
+        chat_template_choice = cfg.get("chat_template") or "chatml"
+        chat_template_jinja = cfg.get("chat_template_jinja")
+    return chat_template_choice, chat_template_jinja
```

Comment:
Is this default to `chatml` a good default? Wouldn't a proper default be the tokenizer's current template?

Reply:
We can change the default to tokenizer_default. Kept it same as before till we reached a decision

Reply:
Was the prior default, chatml? I think the tokenizer's one should be best